camayoc/ui/models/pages/scans.py

```diff
@@ -12,14 +15,19 @@
 
 class ScanListElem(AbstractListItem):
     def download_scan(self) -> Download:
-        scan_locator = "div.list-view-pf-actions button[title=Download]"
-        # timeout argument below is in milliseconds; this is how long
-        # script will wait for download button to appear
-        with self.locator.page.expect_download() as download_info:
-            self.locator.locator(scan_locator).click(timeout=30_000)
-        download = download_info.value
-        download.path()  # blocks the script while file is downloaded
-        return download
+        scan_locator = "td.pf-c-table__action button[data-ouia-component-id=download]"
+        timeout_start = time.monotonic()
+        while 10 * 60 > (time.monotonic() - timeout_start):
+            try:
+                with self.locator.page.expect_download() as download_info:
+                    self.locator.locator(scan_locator).click(timeout=10_000)
```

Comment:
For our automated tests like this, do you prefer hard-coding numbers like this into the functions? Or would these be better served as pseudo-constant variable definitions at the top of the module or project? (or configurable as env vars, for that matter?)

I'm not saying this needs to change; I'm just asking because I'm less familiar with camayoc's preferred coding style conventions. I don't mean to over-complicate this if it's not necessary. 🙂

Reply:
Looking at other timeouts, we aren't very consistent. Sometimes we have values set at function parameter level, sometimes variables at function/class level, sometimes hardcoded like here... The only thing we don't really do is having timeout as global read from config / environment.

I guess there could be some marginal value in some global like "how much time is needed for a scan to complete", but I would probably let it slip until it becomes larger issue.